pkg/asset/installconfig/aws/ec2.go

```diff
@@ -5,11 +5,33 @@ import (
 	"fmt"
 	"time"
 
+	awsv2 "github.com/aws/aws-sdk-go-v2/aws"
+	ec2v2 "github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
 )
 
+func getEC2Client(cfg awsv2.Config) *ec2v2.Client {
+	return ec2v2.NewFromConfig(cfg)
+}
+
+// GetRegions get all regions that are accessible.
+func GetRegions(_ context.Context, config awsv2.Config, allRegions bool) ([]string, error) {
+	client := getEC2Client(config)
+
+	output, err := client.DescribeRegions(context.Background(), &ec2v2.DescribeRegionsInput{AllRegions: aws.Bool(allRegions)})
```

Comment:
This function has a context passed in, you can use that instead of context.Background()